src/components/chats/ChatList/hooks/useGetChatElement.ts

```diff
@@ -0,0 +1,130 @@
+import useWrapCallbackInRef from '@/hooks/useWrapCallbackInRef'
+import { useIsAnyQueriesLoading } from '@/subsocial-query'
+import { generateManuallyTriggeredPromise } from '@/utils/promise'
+import { PostData } from '@subsocial/api/types'
+import { UseQueryResult } from '@tanstack/react-query'
+import { useEffect, useMemo, useRef } from 'react'
+import { getChatItemId } from '../../helpers'
+
+export default function useGetChatElement(
+  commentIds: string[],
+  commentsQuery: UseQueryResult<PostData | null, unknown>[],
+  loadedCommentsQuery: UseQueryResult<PostData | null, unknown>[],
+  loadMore: () => void
+) {
+  const waitAllCommentsLoaded = useWaitCommentsLoading(commentsQuery)
+
+  const waitAllCommentsLoadedRef = useWrapCallbackInRef(waitAllCommentsLoaded)
+  const commentIdsRef = useWrapCallbackInRef(commentIds)
+
+  const promiseRef = useRef<{
+    resolvers: Map<string, VoidFunction[]>
+    waitingCommentIds: Set<string>
+  }>({ resolvers: new Map(), waitingCommentIds: new Set() })
+
+  const loadedCommentIds = useMemo(() => {
+    const ids: string[] = []
+    loadedCommentsQuery.forEach((comment) => {
+      if (comment.data?.id) {
+        ids.push(comment.data.id)
+      }
+    })
+    return ids
+  }, [loadedCommentsQuery])
+
+  const awaitableLoadMore = useAwaitableLoadMore(
+    loadMore,
+    loadedCommentIds.length
+  )
+
+  useEffect(() => {
+    const { waitingCommentIds } = promiseRef.current
+    waitingCommentIds.forEach((commentId) => {
+      if (loadedCommentIds.includes(commentId)) {
+        const resolvers = promiseRef.current.resolvers.get(commentId)
```

Comment:
resolvers is few promises of comments ids that need to load before replied comment, right?

Reply:
yes, and its an array because this hook returns a function,
where if the function is invoked with the same commentId multiple times, I need to resolve all of the promises from those calls